### DISCUSSION TITLE
After installation, test email stuck in queue (pending)what should I do

### DISCUSSION THREAD

**Discussion by xcxcadc** [postalserver/postal#2605] [Help with using Postal]
I use postal status to get the following data，
 SERVICE             CREATED             STATUS              PORTS
postal-cron-1       ghcr.io/postalserver/postal:2.1.4   "/docker-entrypoint.…"   cron                26 minutes ago      Up 26 minutes       
postal-smtp-1       ghcr.io/postalserver/postal:2.1.4   "/docker-entrypoint.…"   smtp                26 minutes ago      Up 26 minutes       
postal-web-1        ghcr.io/postalserver/postal:2.1.4   "/docker-entrypoint.…"   web                 26 minutes ago      Up 26 minutes    
I found that there are two databases that are not running, but when I use docker ps to check, I found the following situation, how can I operate it
TED             STATUS             PORTS                                                                      NAMES
aa522d86b3c5   ghcr.io/postalserver/postal:2.1.4   "/docker-entrypoint.…"   30 minutes ago      Up 30 minutes                                                                                 postal-smtp-1
18f1b9a2be67   ghcr.io/postalserver/postal:2.1.4   "/docker-entrypoint.…"   30 minutes ago      Up 30 minutes                                                                                 postal-cron-1
aba27deccf7f   ghcr.io/postalserver/postal:2.1.4   "/docker-entrypoint.…"   30 minutes ago      Up 30 minutes                                                                                 postal-web-1
9e8439a85456   caddy                               "caddy run --config …"   53 minutes ago      Up 53 minutes                                                                                 postal-caddy
695cd549407b   rabbitmq:3.8                        "docker-entrypoint.s…"   About an hour ago   Up About an hour   4369/tcp, 5671/tcp, 15691-15692/tcp, 25672/tcp, 127.0.0.1:5672->5672/tcp   postal-rabbitmq
1d47d142d44a   mariadb                             "docker-entrypoint.s…"   About an hour ago   Up About an hour   127.0.0.1:3306->3306/tcp                                                   postal-mariadb

**Comment by willpower232** [comment]
Judging from https://github.com/postalserver/install/blob/main/templates/docker-compose.yml it looks like you're missing the worker and requeuer which are responsible for keeping emails sending.

If they work when you `postal start` again then it may be that the rabbitmq docker container wasn't ready when they tried to start so you would need to set up the built in "wait for" feature along these lines: https://github.com/postalserver/postal/discussions/1562#discussioncomment-1275794

**Comment by xcxcadc** [comment]
When I follow this tutorial, he is currently in the following state. I'm sorry for taking s
![5](https://github.com/postalserver/postal/assets/113344244/552fbb21-4ded-43f0-b719-c63b1efd7846)
o long to reply to you. I need your help. Thank you!
[](url)

**Comment by xcxcadc** [comment]
![2](https://github.com/postalserver/postal/assets/113344244/5130e9ce-d47b-4493-ba50-5df9d9a84999)
![1](https://github.com/postalserver/postal/assets/113344244/dc30ed8b-a54a-4f61-b947-7c4b716d1915)
![3](https://github.com/postalserver/postal/assets/113344244/51610264-dc9d-4fd8-a58a-c5adfd576fe8)
In the end, the result is still like this, it cannot run normally. I am using Ubuntu 20.04. The reference for the setup tutorial is: https://technicalsahil.com/how-to-install-configure-postal-mail-server-on -ubuntu/ I don't think I'm getting an error, but I'm weird!

**Comment by xcxcadc** [comment]
When I tear it down and rebuild it, the picture is shown when initializing the database. What should I do?
general:
  # This can be changed to allow messages to be sent from multiple IP addresses
  use_ip_pools: false

web:
  # The host that the management interface will be available on
  host: postal.qwers.live
  # The protocol that requests to the management interface should happen on
  protocol: https

web_server:
  # Specify configuration for the Postal web server
  bind_address: 103.186.65.00
  port: 5000

smtp_server:
  # Specify configuration to the Postal SMTP server
  port: 25

logging:
  # Specify options for the logging
  stdout: true

main_db:
  # Specify the connection details for your MySQL database
  host: 127.0.0.1
  username: root
  password: your_strong_password
  database: posttal

message_db:
  # Specify the connection details for your MySQL server that will be house the
  # message databases for mail servers.
  host: 127.0.0.1
  username: root
  password: your_strong_password
  prefix: postal

rabbitmq:
  # Specify connection details for your RabbitMQ server
  host: 127.0.0.1
  username: postal
  password: postal
  vhost: postalvhost

dns:
  # Specify the DNS records that you have configured. Refer to the documentation at
  # https://github.com/atech/postal/wiki/Domains-&-DNS-Configuration for further
  # information about these.
  mx_records:
    - post.qwers.live
  smtp_server_hostname: postal.qwers.live
  spf_include: spf.postal.qwers.live
  return_path: rp.postal.qwers.live
  route_domain: routes.postal.qwers.live
  track_domain: track.postal.qwers.live

smtp:
  # Specify an SMTP server that can be used to send messages from the Postal management
  # system to users. You can configure this to use a Postal mail server once the
  # your installation has been set up.
  host: 127.0.0.1
  port: 2525
  username: # Complete when Postal is running and you can
  password: # generate the credentials within the interface.
  from_name: Postal
  from_address: postal@qwers.live

rails:
  # This is generated automatically by the config initialization. It should be a random
  # string unique to your installation.
  secret_key: a66cc9db13817df6ad3b5506cd8231603766719f85b2abd88327ceeb56a6b78a6e3fff003e79d91264cc86e13d661336bc33a0b115d34b5581ac62f8ebcbe6ced241a179756a9a8f7eb9ba5029e1505654875939e10fef87744a92668e85342612b32da5377260b10e611f50bbfd582feccf36381abadf5325871e2c64b16fa9

![56](https://github.com/postalserver/postal/assets/113344244/082806ca-b101-4177-b12d-8dbb68c701c9)

**Reply by willpower232** [reply]
It means that you didn't fully remove the original database, things do persist in Docker sometimes.

If you do `docker ps -a` do you see the stopped requeuer and worker? If so, you could try `docker logs` on those two specifically and see what the output is before they stopped working.
